src/shared-components/alert/index.tsx

```diff
@@ -24,7 +24,7 @@ export interface AlertProps {
   content: React.ReactNode;
   ctaContent?: React.ReactNode;
   duration?: string | number;
-  onExit?: ((rest: Omit<AlertProps, 'onExit'>) => void) | (() => void);
+  onExit?: () => void;
```

Comment:
Breaking change, will release as a new major: this `onExit` usage with the rest of the props is no longer used and does not really make sense either (since the component usage has access to all of the props). Including all `alertProps` in the effect was a dependency nightmare so this cleans that up.

All consumer usage should also pass in `onExit` with referential integrity but since the Alert is not given to being re-rendered it is not a big deal if it is not.